### PR TITLE
Caution flag filter front end changes

### PR DESCRIPTION
**Pull request by ksantiagoBAH**
## Description
As a Comparison Tool user, I need to be able to exclude institutions with caution flags so that I can only look at schools without warnings in the search results.

backend changes:
https://github.com/department-of-veterans-affairs/gibct-data-service/pull/635
database migration pr:
https://github.com/department-of-veterans-affairs/gibct-data-service/pull/636
story:
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/7185
## Testing done
1. run database migration from PR .
2. seed your database.
3. open up the rails console type `rails console` in gids terminal.
4. run this in the rails console `Institution.reset_counters(institutions.id, :caution_flags)}` to populate the `count_of_caution_flags` column.
5. There are currently no vettec schools with caution flags so update the `count_of_caution_flags` column in institutions in your database, select any vettec school. 
6. Toggle filters

## Screenshots
Screen Shot 2020-04-03 at 8 36 55 AM
Screen Shot 2020-04-03 at 8 37 05 AM

Screen Shot 2020-04-02 at 8 14 01 AM
VETTEC
Screen Shot 2020-04-03 at 8 49 44 AM
Screen Shot 2020-04-03 at 8 49 54 AM

Screen Shot 2020-04-02 at 8 12 38 AM

## Acceptance criteria
1 . A filter is available on the search results screen that meets the following criteria:
- [x] It is available on the traditional search results page.
- [x] It is available on the VET TEC search results page.
- [x] When selected, it removes institutions with caution flags from the search results page
- [x] When de-selected, it displays institutions with caution flags on the search results page that meet all other search criteria.
- [x] It has a Learn More Link to display additional content to the user.
When clicking on the Learn more link (1e), the following information is displayed:

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/gi/components/search/InstitutionFilterForm.jsx [hunk 1]
```diff
@@ -67,6 +78,25 @@ class InstitutionFilterForm extends React.Component {
     );
   };
 
+  renderCautionaryWarningsFilter = () => {
```
Comment (zurbergram) [inline src/applications/gi/components/search/InstitutionFilterForm.jsx:81]: can you break this out into its own component like how https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/gi/components/search/OnlineClassesFilter.jsx is ?

File: src/applications/gi/components/vet-tec/VetTecSearchForm.jsx [hunk 2]
```diff
@@ -115,6 +132,22 @@ class VetTecSearchForm extends React.Component {
     );
   };
 
+  renderCautionaryWarningsFilter = () => {
```
Comment (zurbergram) [inline src/applications/gi/components/vet-tec/VetTecSearchForm.jsx:135]: can you break this out into its own component like how https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/gi/components/search/OnlineClassesFilter.jsx is ?

Comment (ksantiagomorgan6) [reply src/applications/gi/components/vet-tec/VetTecSearchForm.jsx:135]: Sure

**Review by sonntag-adam — APPROVED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Pull request merged**